File: basic/src/main/java/com/yuanshenbin/basic/state/NetworkState.java
package com.yuanshenbin.basic.state;

import android.support.annotation.DrawableRes;
import android.support.annotation.Nullable;

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 网络异常状态
 */

public abstract class NetworkState extends StateAbstract {
    protected OnRetryListener listener;
    
    public NetworkState(@Nullable OnRetryListener listener) {
        this.listener = listener;
    }

    public abstract void showNetwork(@DrawableRes int resId, String... msg);

    public abstract void showNetwork(@DrawableRes int resId);

    public abstract void showNetwork(String... msg); 
    
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/state/NetworkState.kt
package com.yuanshenbin.basic.state

import androidx.annotation.DrawableRes

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 网络异常状态
 */
abstract class NetworkState(open var listener: OnRetryListener?) : StateAbstract() {

    abstract fun showNetwork(@DrawableRes resId: Int, vararg msg: String?)
    abstract fun showNetwork(@DrawableRes resId: Int)
    abstract fun showNetwork(vararg msg: String?)

}